BLD: build warnings from quadpack

### Describe your issue.

The recent Quadpack rewrite in C results in a few probably harmless build warnings. Opening the issue for visbility mostly. CC @ilayn 

### Reproducing Code Example

```python
[1074/1277] Compiling C object scipy/integrate/_quadpack.cpython-312-x86_64-linux-gnu.so.p/__quadpack.c.o
../scipy/integrate/__quadpack.c: In function 'dqagie':
../scipy/integrate/__quadpack.c:540:16: warning: 'erlarg' may be used uninitialized [-Wmaybe-uninitialized]
  540 |         erlarg = erlarg - erlast;
      |         ~~~~~~~^~~~~~~~~~~~~~~~~
../scipy/integrate/__quadpack.c:406:34: note: 'erlarg' was declared here
  406 |     double defab1, defab2, dres, erlarg, erlast, errbnd, errmax, error1, error2;
      |                                  ^~~~~~
../scipy/integrate/__quadpack.c:551:27: warning: 'ertest' may be used uninitialized [-Wmaybe-uninitialized]
  551 |         if ((ierror == 3) || (erlarg <= ertest)) { goto LINE60; }
      |             ~~~~~~~~~~~~~~^~~~~~~~~~~~~~~~~~~~~
../scipy/integrate/__quadpack.c:407:29: note: 'ertest' was declared here
  407 |     double error12, errsum, ertest, resabs, reseps, small;
      |                             ^~~~~~
../scipy/integrate/__quadpack.c: In function 'dqagse':
../scipy/integrate/__quadpack.c:1529:27: warning: 'ertest' may be used uninitialized [-Wmaybe-uninitialized]
 1529 |         if ((ierror == 3) || (erlarg <= ertest)) { goto LINE60; }
      |             ~~~~~~~~~~~~~~^~~~~~~~~~~~~~~~~~~~~
../scipy/integrate/__quadpack.c:1386:12: note: 'ertest' was declared here
 1386 |     double ertest, small, jupbnd, correc, resabs, reseps, abseps;
      |            ^~~~~~
```


### Error message

```shell
N/A
```


### SciPy/NumPy/Python version and system information

```shell
1.15.0.dev0+git20241008.548350a 2.1.2 sys.version_info(major=3, minor=12, micro=7, releaselevel='final', serial=0)
Build Dependencies:
  blas:
    detection method: pkgconfig
    found: true
    include directory: /home/tyrion/miniforge3/envs/scipy-dev/include
    lib directory: /home/tyrion/miniforge3/envs/scipy-dev/lib
    name: openblas
    openblas configuration: USE_64BITINT=0 DYNAMIC_ARCH=1 DYNAMIC_OLDER= NO_CBLAS=
      NO_LAPACK=0 NO_LAPACKE= NO_AFFINITY=1 USE_OPENMP=0 PRESCOTT MAX_THREADS=128
    pc file directory: /home/tyrion/miniforge3/envs/scipy-dev/lib/pkgconfig
    version: 0.3.27
  lapack:
    detection method: pkgconfig
    found: true
    include directory: /home/tyrion/miniforge3/envs/scipy-dev/include
    lib directory: /home/tyrion/miniforge3/envs/scipy-dev/lib
    name: openblas
    openblas configuration: USE_64BITINT=0 DYNAMIC_ARCH=1 DYNAMIC_OLDER= NO_CBLAS=
      NO_LAPACK=0 NO_LAPACKE= NO_AFFINITY=1 USE_OPENMP=0 PRESCOTT MAX_THREADS=128
    pc file directory: /home/tyrion/miniforge3/envs/scipy-dev/lib/pkgconfig
    version: 0.3.27
  pybind11:
    detection method: pkgconfig
    include directory: /home/tyrion/miniforge3/envs/scipy-dev/include
    name: pybind11
    version: 2.13.6
Compilers:
  c:
    args: -march=nocona, -mtune=haswell, -ftree-vectorize, -fPIC, -fstack-protector-strong,
      -fno-plt, -O2, -ffunction-sections, -pipe, -isystem, /home/tyrion/miniforge3/envs/scipy-dev/include,
      -DNDEBUG, -D_FORTIFY_SOURCE=2, -O2, -isystem, /home/tyrion/miniforge3/envs/scipy-dev/include
    commands: /home/tyrion/miniforge3/envs/scipy-dev/bin/x86_64-conda-linux-gnu-cc
    linker: ld.bfd
    linker args: -Wl,-O2, -Wl,--sort-common, -Wl,--as-needed, -Wl,-z,relro, -Wl,-z,now,
      -Wl,--disable-new-dtags, -Wl,--gc-sections, -Wl,--allow-shlib-undefined, -Wl,-rpath,/home/tyrion/miniforge3/envs/scipy-dev/lib,
      -Wl,-rpath-link,/home/tyrion/miniforge3/envs/scipy-dev/lib, -L/home/tyrion/miniforge3/envs/scipy-dev/lib,
      -march=nocona, -mtune=haswell, -ftree-vectorize, -fPIC, -fstack-protector-strong,
      -fno-plt, -O2, -ffunction-sections, -pipe, -isystem, /home/tyrion/miniforge3/envs/scipy-dev/include,
      -DNDEBUG, -D_FORTIFY_SOURCE=2, -O2, -isystem, /home/tyrion/miniforge3/envs/scipy-dev/include
    name: gcc
    version: 13.3.0
  c++:
    args: -fvisibility-inlines-hidden, -fmessage-length=0, -march=nocona, -mtune=haswell,
      -ftree-vectorize, -fPIC, -fstack-protector-strong, -fno-plt, -O2, -ffunction-sections,
      -pipe, -isystem, /home/tyrion/miniforge3/envs/scipy-dev/include, -DNDEBUG, -D_FORTIFY_SOURCE=2,
      -O2, -isystem, /home/tyrion/miniforge3/envs/scipy-dev/include
    commands: /home/tyrion/miniforge3/envs/scipy-dev/bin/x86_64-conda-linux-gnu-c++
    linker: ld.bfd
    linker args: -Wl,-O2, -Wl,--sort-common, -Wl,--as-needed, -Wl,-z,relro, -Wl,-z,now,
      -Wl,--disable-new-dtags, -Wl,--gc-sections, -Wl,--allow-shlib-undefined, -Wl,-rpath,/home/tyrion/miniforge3/envs/scipy-dev/lib,
      -Wl,-rpath-link,/home/tyrion/miniforge3/envs/scipy-dev/lib, -L/home/tyrion/miniforge3/envs/scipy-dev/lib,
      -fvisibility-inlines-hidden, -fmessage-length=0, -march=nocona, -mtune=haswell,
      -ftree-vectorize, -fPIC, -fstack-protector-strong, -fno-plt, -O2, -ffunction-sections,
      -pipe, -isystem, /home/tyrion/miniforge3/envs/scipy-dev/include, -DNDEBUG, -D_FORTIFY_SOURCE=2,
      -O2, -isystem, /home/tyrion/miniforge3/envs/scipy-dev/include
    name: gcc
    version: 13.3.0
  cython:
    commands: cython
    linker: cython
    name: cython
    version: 3.0.11
  fortran:
    args: -march=nocona, -mtune=haswell, -ftree-vectorize, -fPIC, -fstack-protector-strong,
      -fno-plt, -O2, -ffunction-sections, -pipe, -isystem, /home/tyrion/miniforge3/envs/scipy-dev/include
    commands: /home/tyrion/miniforge3/envs/scipy-dev/bin/x86_64-conda-linux-gnu-gfortran
    linker: ld.bfd
    linker args: -Wl,-O2, -Wl,--sort-common, -Wl,--as-needed, -Wl,-z,relro, -Wl,-z,now,
      -Wl,--disable-new-dtags, -Wl,--gc-sections, -Wl,--allow-shlib-undefined, -Wl,-rpath,/home/tyrion/miniforge3/envs/scipy-dev/lib,
      -Wl,-rpath-link,/home/tyrion/miniforge3/envs/scipy-dev/lib, -L/home/tyrion/miniforge3/envs/scipy-dev/lib,
      -march=nocona, -mtune=haswell, -ftree-vectorize, -fPIC, -fstack-protector-strong,
      -fno-plt, -O2, -ffunction-sections, -pipe, -isystem, /home/tyrion/miniforge3/envs/scipy-dev/include
    name: gcc
    version: 13.3.0
  pythran:
    include directory: ../../../miniforge3/envs/scipy-dev/lib/python3.12/site-packages/pythran
    version: 0.16.1
Machine Information:
  build:
    cpu: x86_64
    endian: little
    family: x86_64
    system: linux
  cross-compiled: false
  host:
    cpu: x86_64
    endian: little
    family: x86_64
    system: linux
Python Information:
  path: /home/tyrion/miniforge3/envs/scipy-dev/bin/python3.12
  version: '3.12'
```

--- a/scipy/integrate/__quadpack.c
+++ b/scipy/integrate/__quadpack.c
@@ -419,6 +419,9 @@ dqagie(double(*fcn)(double* x), const double bound, const int inf,
     rlist[0] = 0.0;
     elist[0] = 0.0;
     iord[0] = 0;
+    // Initialize variables to silence compiler
+    erlarg = 0.0;
+    ertest = 0.0;  
     if ((epsabs <= 0.0) && (epsrel < fmax(50.0*epmach, 0.5e-28))) { *ier = 6; }
     if (*ier == 6) { return; }
 
@@ -1399,6 +1402,7 @@ dqagse(double(*fcn)(double* x), const double a, const double b,
     *abserr = 0.0;
     ierror = 0;
     erlarg = 0.0;
+    ertest = 0.0;
     correc = 0.0;
 
     if ((epsabs <= 0.0) && (epsrel < fmax(50.0*epmach, 0.5e-28)))

